pkg/apis/operator/v1alpha1/glustercluster_types.go

```diff
@@ -1,22 +1,60 @@
 package v1alpha1
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+// GlusterStorageTarget defines a storage target
+type GlusterStorageTarget struct {
+	Name    string      `json:"name"`
+	Address []string    `json:"address"`
+	Creds   Credentials `json:"credentials"`
+}
+
+// GlusterClusterReplicationDetails defines replication details
+type GlusterClusterReplicationDetails struct {
+	Creds   Credentials            `json:"credentials"`
+	Targets []GlusterStorageTarget `json:"targets"`
+}
+
+// GlusterNodeThreshold defines threshold details for node
+type GlusterNodeThreshold struct {
+	Nodes          int    `json:"nodes"`
```

Comment:
`Nodes`, `MinNodes`, and `MaxNodes` need to be `*int` so omitempty can differentiate between 0 and absent

Reply:
Ack